packages/assets-controllers/src/MultichainBalancesController/MultichainBalancesController.ts

```diff
@@ -174,22 +176,73 @@ export class MultichainBalancesController extends BaseController<
       (balanceUpdate: AccountBalancesUpdatedEventPayload) =>
         this.#handleOnAccountBalancesUpdated(balanceUpdate),
     );
-    // TODO: Maybe add a MultichainAssetsController:accountAssetListUpdated event instead of using the entire state.
-    // Since MultichainAssetsController already listens for the AccountsController:accountAdded, we can rely in it for that event
-    // and not listen for it also here, in this controller, since it would be redundant
+
     this.messagingSystem.subscribe(
-      'MultichainAssetsController:stateChange',
-      async (assetsState: MultichainAssetsControllerState) => {
-        for (const accountId of Object.keys(assetsState.accountsAssets)) {
-          await this.#updateBalance(
-            accountId,
-            assetsState.accountsAssets[accountId],
-          );
-        }
+      'MultichainAssetsController:newAccountAssets',
+      async ({ newAccountAssets }) => {
+        await this.#updateBalancesForAccounts(newAccountAssets);
       },
     );
   }
 
+  /**
+   * Updates the balances for the given accounts.
+   *
+   * @param accounts - The accounts to update the balances for.
+   */
+  async #updateBalancesForAccounts(
+    accounts: {
+      accountId: string;
+      assets: CaipAssetType[];
+    }[],
+  ): Promise<void> {
+    const { isUnlocked } = this.messagingSystem.call(
+      'KeyringController:getState',
+    );
+
+    if (!isUnlocked) {
+      return;
+    }
```

Comment:
Nit: I think that's a good thing to add here! We could wrap this in a method that uses a callback, something like:

```ts
withUnlocked(() => {
   const balancesToUpdate: MultichainBalancesControllerState['balances'] = {};
   
   ...
});
```

Reply:
Thanks Charly 🙏  resolving this for now as we are trying to make less changes when possible 🙏